src/protocols/secure_channel/MessageCounterManager.cpp

```diff
@@ -225,6 +225,9 @@ CHIP_ERROR MessageCounterManager::SendMsgCounterSyncResp(Messaging::ExchangeCont
 
     VerifyOrDie(exchangeContext->HasSessionHandle());
 
+    VerifyOrReturnError(exchangeContext->GetSessionHandle()->GetSessionType() == Session::SessionType::kSecure,
+                        CHIP_ERROR_INVALID_ARGUMENT);
```

Comment:
But an MCSP request in fact uses a group session per spec, so this check would always fail for a valid MCSP request....

Reply:
MCSP is not currently implemented for groups - otherwise this function would work.

From the group messaging discussions, it seems MCSP missing is a known issue and only trust-first is to be enabled.

This check is correct given existing code: this code can only work if the session is secure, based on the cast a few lines below.

Reply:
4.9.2.5. Message Counter Synchronization Session Context actually has additional metadata.  So yes we will use the operational group key hash as group ID.  But I would think that were this to be implemented, we would want and need to be seeing a completely distinct MCSP session type here.

Reply:
Updated content and added a comment. It seems we want to implement MCSP and existing "counters for secure sessions" should not be a thing (secure sessions should create their own counters).